data/en/test_routing_values_array.json

```diff
@@ -0,0 +1,279 @@
+                            }]
+                        }]
+                    },
+                    {
+                        "id": "block-three",
+                        "type": "Question",
+                        "questions": [{
+                            "id": "question-three",
+                            "title": "Page-three",
+                            "type": "General",
+                            "answers": [{
+                                "id": "answer-three",
+                                "mandatory": false,
+                                "type": "Number",
+                                "label": "Answer-three",
+                                "decimal_places": 0
+                            }]
+                        }]
+                    },
+                    {
+                        "id": "block-four",
+                        "type": "Question",
+                        "questions": [{
+                            "id": "question-four",
+                            "title": "Page-four",
+                            "type": "General",
+                            "answers": [{
+                                "id": "answer-four",
+                                "mandatory": false,
+                                "type": "Checkbox",
+                                "label": "not all of",
+                                "options": [{
+                                        "label": "Box1",
+                                        "value": "Box1"
+                                    },
+                                    {
+                                        "label": "Box2",
+                                        "value": "Box2"
+                                    },
+                                    {
+                                        "label": "Box3",
+                                        "value": "Box3"
+                                    },
+                                    {
+                                        "label": "Box4",
+                                        "value": "Box4"
+                                    }
+                                ]
+                            }]
+                        }],
+                        "routing_rules": [{
+                            "goto": {
+                                "block": "block-five",
+                                "when": [{
+                                    "id": "answer-four",
+                                    "condition": "not all of",
+                                    "values": ["Box1", "Box2", "Box3"]
+                                }]
+                            }
+                        }, {
+                            "goto": {
+                                "block": "block-six",
+                                "when": [{
+                                    "id": "answer-four",
+                                    "condition": "not any of",
```

Comment:
Is this being tested? Is this rule only invoked if previous "not all of" rule is false i.e. if Box1, Box2, Box3 all selected.

Reply:
Good spot, its being testing on block 6 I forgot to delete it from here when I moved it.